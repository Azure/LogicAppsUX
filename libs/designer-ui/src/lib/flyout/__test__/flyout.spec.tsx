import type { FlyoutProps } from '../index';
import { Flyout } from '../index';
import renderer from 'react-test-renderer';

describe('ui/flyout', () => {
  let minimal: FlyoutProps;

  beforeEach(() => {
    minimal = {
      text: 'text',
    };
  });

  it('should render', () => {
    const tree = renderer.create(<Flyout {...minimal} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with an ARIA label', () => {
    const tree = renderer.create(<Flyout {...minimal} ariaLabel="aria-label" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with a non-zero tab index', () => {
    const tree = renderer.create(<Flyout {...minimal} tabIndex={-1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with a title', () => {
    const tree = renderer.create(<Flyout {...minimal} title="title" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
