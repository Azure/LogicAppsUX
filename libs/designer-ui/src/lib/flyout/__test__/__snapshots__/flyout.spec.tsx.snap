// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/flyout should render 1`] = `
<div
  className="ms-TooltipHost root-116"
  onBlurCapture={[Function]}
  onFocusCapture={[Function]}
  onKeyDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="none"
>
  <button
    aria-label="text"
    className="msla-button msla-flyout"
    data-testid="callout-btn"
    onClick={[Function]}
    tabIndex={0}
  >
    <img
      alt=""
      className="msla-flyout-icon"
      draggable={false}
      role="presentation"
      src="information_tiny.svg"
    />
  </button>
  <div
    hidden={true}
    id="tooltip0"
    style={
      {
        "border": 0,
        "height": 1,
        "margin": -1,
        "overflow": "hidden",
        "padding": 0,
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": 1,
      }
    }
  />
</div>
`;

exports[`ui/flyout should render with a non-zero tab index 1`] = `
<div
  className="ms-TooltipHost root-116"
  onBlurCapture={[Function]}
  onFocusCapture={[Function]}
  onKeyDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="none"
>
  <button
    aria-label="text"
    className="msla-button msla-flyout"
    data-testid="callout-btn"
    onClick={[Function]}
    tabIndex={-1}
  >
    <img
      alt=""
      className="msla-flyout-icon"
      draggable={false}
      role="presentation"
      src="information_tiny.svg"
    />
  </button>
  <div
    hidden={true}
    id="tooltip2"
    style={
      {
        "border": 0,
        "height": 1,
        "margin": -1,
        "overflow": "hidden",
        "padding": 0,
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": 1,
      }
    }
  />
</div>
`;

exports[`ui/flyout should render with a title 1`] = `
<div
  className="ms-TooltipHost root-116"
  onBlurCapture={[Function]}
  onFocusCapture={[Function]}
  onKeyDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="none"
>
  <button
    aria-label="title"
    className="msla-button msla-flyout"
    data-testid="callout-btn"
    onClick={[Function]}
    tabIndex={0}
  >
    <img
      alt=""
      className="msla-flyout-icon"
      draggable={false}
      role="presentation"
      src="information_tiny.svg"
    />
  </button>
  <div
    hidden={true}
    id="tooltip3"
    style={
      {
        "border": 0,
        "height": 1,
        "margin": -1,
        "overflow": "hidden",
        "padding": 0,
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": 1,
      }
    }
  />
</div>
`;

exports[`ui/flyout should render with an ARIA label 1`] = `
<div
  className="ms-TooltipHost root-116"
  onBlurCapture={[Function]}
  onFocusCapture={[Function]}
  onKeyDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="none"
>
  <button
    aria-label="aria-label"
    className="msla-button msla-flyout"
    data-testid="callout-btn"
    onClick={[Function]}
    tabIndex={0}
  >
    <img
      alt=""
      className="msla-flyout-icon"
      draggable={false}
      role="presentation"
      src="information_tiny.svg"
    />
  </button>
  <div
    hidden={true}
    id="tooltip1"
    style={
      {
        "border": 0,
        "height": 1,
        "margin": -1,
        "overflow": "hidden",
        "padding": 0,
        "position": "absolute",
        "whiteSpace": "nowrap",
        "width": 1,
      }
    }
  >
    aria-label
  </div>
</div>
`;
